### PR TITLE
Deprecate http usage: `delete` operations, promote warning to error for http sources

### PR DESCRIPTION
**Pull request by Nigusu-Allehu**
## Bug

Fixes: https://github.com/NuGet/Home/issues/13341

Regression? Last working version:

## Description

Show an error when the `delete` operation is used with an http source
## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs [hunk 1]
```diff
@@ -761,6 +761,7 @@ private PackageDependency CreateDependencyFromProject(dynamic project, Dictionar
 
                 projectFactory.InitializeProperties(builder);
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs]: I think you need to rebase the feature branch and then rebase this branch off of the feature branch.

This commit shouldn't be part of this PR.

**Comment by Nigusu-Allehu** [comment]
@NuGet/nuget-client any suggestions?

**Review by nkolev92 — APPROVED** [review]
